service/provider/src/main/java/gov/va/vro/service/provider/MasPollingProcessor.java

```diff
@@ -1,42 +1,156 @@
-
   private final CamelEntrance camelEntrance;
   private final MasDelays masDelays;
+  private final MasApiService masCollectionAnnotsApiService;
+  private final ObjectMapper objectMapper = new ObjectMapper();
 
   @Override
+  @SneakyThrows
   public void process(Exchange exchange) {
-    var payload = exchange.getMessage().getBody(MasAutomatedClaimPayload.class);
-    log.info("Checking collection status for collection {}.", payload.getCollectionId());
-    // call pcCheckCollectionStatus
-    boolean ready = checkCollectionStatus(payload.getCollectionId());
-    if (ready) {
-      log.info("Collection {} is ready for processing.", payload.getCollectionId());
-      // call pcQueryCollectionAnnots
+
+    // TODO : Remove the extraneous log statements
+    var claimPayload = exchange.getMessage().getBody(MasAutomatedClaimPayload.class);
+
+    boolean isCollectionReady = checkCollectionStatus(claimPayload);
+
+    if (isCollectionReady) {
+      AbdEvidence abdEvidence = getCollectionAnnots(claimPayload);
       // call Lighthouse
       // Combine results and call PDF generation
-      // if a decision is made, call pcOrderExam
+      GeneratePdfResp generatePdfResp = generatePdf(claimPayload, abdEvidence);
+      // call pcOrderExam in the absence of evidence
     } else {
-      log.info("Collection {} is not ready. Requeueing...", payload.getCollectionId());
+      log.info("Collection {} is not ready. Requeue..ing...", claimPayload.getCollectionId());
       // re-request after some time
-      camelEntrance.notifyAutomatedClaim(payload, masDelays.getMasProcessingSubsequentDelay());
+      camelEntrance.notifyAutomatedClaim(claimPayload, masDelays.getMasProcessingSubsequentDelay());
+    }
+  }
+
+  public boolean checkCollectionStatus(MasAutomatedClaimPayload claimPayload) throws MasException {
+
+    log.info("Checking collection status for collection {}.", claimPayload.getCollectionId());
+    boolean isCollectionReady = false;
+    try {
+      List<Integer> collectionIds = new ArrayList<Integer>();
+      collectionIds.add(claimPayload.getCollectionId());
+      var response = masCollectionAnnotsApiService.getMasCollectionStatus(collectionIds);
+      log.info("Collection Status Response : response Size: " + response.size());
+      for (MasCollectionStatus masCollectionStatus : response) {
+        log.info(
+            "Collection Status Response : Collection ID  {} ",
+            masCollectionStatus.getCollectionsId());
+        log.info(
+            "Collection Status Response : Collection Status {} ",
+            masCollectionStatus.getCollectionStatus());
+        if ((MasStatus.PROCESSED)
+            .equals(MasStatus.getMasStatus(masCollectionStatus.getCollectionStatus()))) {
+          isCollectionReady = true;
+        }
+      }
+    } catch (Exception e) {
+      log.error("Error in calling collection Status API ", e);
+      throw new MasException(e.getMessage(), e);
     }
+    return isCollectionReady;
```

Comment:
I am not sure how this exception being handled but I will worry about it when this is complete.